### ISSUE TITLE
Many codepoints have the wrong Softbank Emoji

### ISSUE THREAD

**Issue by April93** [iamcal/emoji-data#95]
Pretty much as the title says. Many emoji have an improper softbank code. It's clear that the problem is due to copying a flawed list from [this repo](https://github.com/googlei18n/emoji4unicode) which is no longer maintained. That list improperly maps softbank emoji to multiple unicode characters (with some emoji being listed for multiple unicode values).

A quick example would be SB:E44B, which is *supposed* to be U+1F303 called 'Night with Stars'. If you look at the mapping you can see that this is indeed the case. *However*, SB:E44B is *also* listed for U+1F30C 'MILKY WAY' and U+1F309 'Bridge At Night', both of which are obviously incorrect (Milky Way being an emoji of our galaxy, and Bridge at night being, well, a bridge at night). SB:E44B is clearly meant to be U+1F303, and *not* the other two.

This leads to bad conversions when you actually use the table (resulting in whatever of the three you end up grabbing). This is a recurring problem throughout the set (The red apple emoji being labeled as Green Apple for instance).

Unfortunately the original list is no longer maintained. However, given that this is the most popular/actively updated conversion table, I figured I'd list my issue here.

Is there a proper/fixed list available anywhere? [Unicode.org](http://unicode.org/emoji/charts/full-emoji-list.html) lists the emoji along side their proper codepoint, but lack the softbank values (so you can't make a conversion table). I ripped those properly, but they differ in style from the emoji listed on [Softbank's official website.](http://creation.mb.softbank.jp/mc/tech/tech_pic/pic_material_1.html) So as of now I have the latest updated emoji from softbank's site listed with their SB value, and the older unicode.org set with the proper unicode values.  But I have no way to convert either of them properly.

I might end up just making the conversion table by hand, but I'd rather not have to do that. Besides the duplicates, I'm also worried the other mappings might be wrong (otherwise I'd just remove the duplicates by hand and then continue on). If someone could provide a proper table, or a fixed version of this one, that'd be great :)

**Comment by April93** [comment]
Upon digging, it appears the error stems back to [this KDDI->Softbank/Docomo conversion table](http://trialgoods.com/emoji/?career=au&page=all), which indeed has both the problems mentioned above (starry night/galaxy/etc and green->red apple).

The author of the page provides a better Softbank -> KDDI conversion table [here.](http://trialgoods.com/emoji/?career=sb&page=all) However, given the age and intent of the page, there's no conversion to proper unicode. ~~Arguably it should be a simple matter of converting Softbank->KDDI->Unicode to arrive at the proper codepoint and name. Along with that it'd be possible to generate a proper Softbank->Unicode table, which I'll upload here once I do it.~~ While the conversion is *better*, there's still an issue of being multiple Softbank emoji for a single KDDI emoji. Ugh.

I'll keep the issue updated as I work.

**Comment by April93** [comment]
Provided a proper table+comments [on the wiki](https://github.com/iamcal/emoji-data/wiki/Proper-Softbank--Unicode-Lookup-Table). The proper table was found on [unicode.org](http://www.unicode.org/Public/UNIDATA/EmojiSources.txt). However this requires the softbank shift-jis values. A mapping of the softbank unicode to softbank shift-jis can be found [here](http://cpansearch.perl.org/src/MIYAGAWA/Encode-JP-Mobile-0.18/ucm/x-utf8-softbank.ucm). And of course the softbank emoji and their softbank unicode values can be found [on the official site.](http://creation.mb.softbank.jp/mc/tech/tech_pic/pic_material_1.html)

The unicode.org site lacked one emoji which I added, and the extra shibuya emoji, which I made a mapping for (since it *is* used in modern emoji sets). The 10 o'clock/alarm clock emoji doesn't correlate to a single emoji if you are using the new softbank set, but *does* if you're using the old one. In the list it's mapped to the 10 o'clock character that fits both.

Lastly, the old softbank set included some more characters, but all were removed from the new set, and are more or less commercial logos, and were not added to unicode. As such, they are not in the table. If you wish to include them in a font, use their existing softbank unicode values.
